sigv4/sigv4.go

```diff
@@ -62,6 +62,7 @@ func NewSigV4RoundTripper(cfg *SigV4Config, next http.RoundTripper) (http.RoundT
 	sess, err := session.NewSessionWithOptions(session.Options{
 		Config: aws.Config{
 			Region:      aws.String(cfg.Region),
+			Endpoint:    aws.String(cfg.Endpoint),
```

Comment:
I think what you want to change is the property `STSRegionalEndpoint` and set it to `RegionalSTSEndpoint`

References:
https://docs.aws.amazon.com/sdk-for-go/api/aws/#Config
https://github.com/aws/aws-sdk-go/blob/main/aws/endpoints/endpoints.go#L186C2-L186C21

Then we don't need to add a configuration because it is fine to always use the regional STS endpoint.

Reply:
On the other hand, I think it is safer to add a configuration parameter whose default value is the current behaviour (global endpoint - value `LegacySTSEndpoint`). In fact you the zero value will be the legacy endpoint so it is fine to not set if the configuration is not setting.

 https://docs.aws.amazon.com/IAM/latest/UserGuide/id_credentials_temp_enable-regions.html#id_credentials_region-endpoints

Reply:
still we have the benefit of not needing to provide the sts endpoint explicitly and we leave the SDK to do that for us.

Reply:
(Responding even though this PR is closed and will be moved)

Note that `STSRegionalEndpoint` is unrelated to the problem at hand here, which is the lack of ability to use a VPC endpoint for STS when one needs to avoid public-facing endpoints altogether.